### DISCUSSION TITLE
Heartbeat misses and closed connections

### DISCUSSION THREAD

**Discussion by RoloEdits** [gftea/amqprs#133] [Q&A] [answered]
I'm having some trouble where my client or the cloudamqp service aren't responding to heartbeats and I get missed heartbeat warnings. The tier that I am using doesn't let me see logs, so all I have to go off of are the client logs. I am using the default 60 seconds.

After #126 this has really become an issue. I've tried searching everywhere and by all means it seems it should just work.

Is there a setup process I missed to set up heartbeats to respond in the library?

**Comment by gftea** [comment]
Have you read the guide https://www.rabbitmq.com/docs/heartbeats

**Reply by RoloEdits** [reply]
I have, and with the default of 60 seconds, when the recommendation is 5-20, I should have been out of the false positive range. I have tried 300 second as well and still the same issue. 

The error message I get is: `missing heartbeat from server for '... [open]'`  and  `received shutdown notification for connection '... [closed]'`

First one makes it seem like its queue instance thing, but the second one is like the queue was fine and the consumer was faulty and so closed the connection. Just trying to narrow what is going on has made no progress, and like I said before, I cant see the logs from the queue, so I only have the client ones I can go off of.

**Reply by gftea** [reply]
I think there is issue in my pr, I will try to fix .

you can downgrade to previous version or try set heartbeat to 0 on client side .

**Reply by gftea** [reply, marked as answer]
@RoloEdits , I have released a new version v1.6.2 which should fix the issue

**Reply by RoloEdits** [reply]
Yes, it definitely helped. Before I was relying entirely on the auto-redeploy to keep my services up, and although I still see the missed heartbeat messages, `missing heartbeat from server for '... [open]'`, way more than I would think I should, it seems to be holding on enough; this brings it inline with where I was before. I would also love to know if this message is proof that its the queue side, where its not returning a response?

I'm not sure if what I have still is another bug somewhere, or if like the heartbeat is expecting an exact timed response and the latency is causing some beats to be off time, as I don't have this problem locally, but I will accept this as the answer. Thanks for the quick response! If I manage to narrow down the issue, and it does lead back to the library, I'll be sure to open an issue.

**Reply by RoloEdits** [reply]
Also I guess for completeness sake I am using the Loyal Lemming plan, the LavinMQ one, from cloudamqp. If you wanted to mess around to test yourself. The services are hosted on digital ocean, the shared App Platform plan, the lowest cost one.
